source/parser/parse_utils.cpp

```diff
@@ -52,4 +56,20 @@ uint32_t parse_with_default(
         return default_value;
     }
     return value_parser(node[field_name].as<std::string>());
-}
\ No newline at end of file
+}
+
+std::unique_ptr<sim::IHasher> parse_hasher(const YAML::Node &node) {
+    const std::string hasher_field_name = "hasher";
+    const std::string hasher_value = node[hasher_field_name].as<std::string>();
+    if (hasher_value == "random") {
+        return std::make_unique<sim::RandomHasher>();
+    }
+    if (hasher_value == "base") {
+        return std::make_unique<sim::BaseHasher>();
+    }
+    if (hasher_value == "symmetric") {
+        return std::make_unique<sim::SymmetricHasher>();
+    }
+    LOG_WARN(fmt::format("Unknown hasher type: {}", hasher_value));
+    return std::make_unique<sim::BaseHasher>();
```

Comment:
I think its better to use `LOG_ERROR` and return `nullptr` here:
```suggestion
    LOG_ERROR(fmt::format("Unknown hasher type: {}", hasher_value));
    return nullptr;
```

Reply:
Do we really want to force user to specify hasher? I think it would be more convenient to just set some default value if hasher is not specified and write some warining log, as suggested [here](https://github.com/Cloud-Storage-Team/algnet/pull/370/files#r2222249119)